### PR TITLE
Tidying up spacetime exports

### PR DESCRIPTION
**Pull request by dra27**
More bits and bobs found while tidying up the globals for Cygwin64

### CODE REVIEW

File: runtime/caml/spacetime.h [hunk 1]
```diff
@@ -195,9 +195,6 @@ extern void caml_spacetime_automatic_snapshot (void);
   profinfo = (uintnat) 0;
 #endif
 
-#else
 
-#define Get_my_profinfo_with_cached_backtrace(profinfo, size)   \
-  profinfo = (uintnat) 0;
```
Comment (dra27) [inline runtime/caml/spacetime.h:201]: This got added in #735 but wasn't properly removed in https://github.com/ocaml/ocaml/pull/1477

File: runtime/spacetime_nat.c [hunk 2]
```diff
@@ -131,7 +129,7 @@ static void reinitialise_free_node_block(void)
 
 enum {
   FEATURE_CALL_COUNTS = 1,
-} features;
+};
```
Comment (dra27) [inline runtime/spacetime_nat.c:132]: Naming the enum makes it global, which isn't what's wanted (I think)

File: runtime/spacetime_nat.c [hunk 3]
```diff
@@ -211,7 +209,8 @@ static void maybe_reopen_snapshot_channel(void)
   }
 }
 
-extern void caml_spacetime_automatic_save(void);
+/* Defined later */
+static void caml_spacetime_automatic_save(void);
```
Comment (dra27) [inline runtime/spacetime_nat.c:213]: This is just a forward definition, so `static` is more appropriate (there are other like it - there are other candidates for this in `spacetime_nat.c` too, but this is the only one which ends up with an `extern` on it...)

File: runtime/spacetime_nat.c [hunk 4]
```diff
@@ -355,8 +354,8 @@ CAMLprim value caml_spacetime_save_event (value v_time_opt,
 }
 
 
-void save_trie (struct channel *chan, double time_override,
-                int use_time_override)
+static void save_trie (struct channel *chan, double time_override,
```
Comment (dra27) [inline runtime/spacetime_nat.c:347]: This gets caught (correctly) by tool/check-symbol-names and shouldn't be global without being prefixed `caml_`

File: runtime/spacetime_nat.c [hunk 5]
```diff
@@ -996,9 +995,6 @@ void caml_spacetime_c_to_ocaml(void* ocaml_entry_point,
     || Is_ocaml_node(*caml_spacetime_trie_node_ptr));
 }
 
-extern void caml_garbage_collection(void);  /* signals_nat.c */
```
Comment (dra27) [inline runtime/spacetime_nat.c:999]: This is defined in `minor_gc.h` (which is already `#include`'d

File: runtime/spacetime_nat.c [hunk 6]
```diff
@@ -996,9 +995,6 @@ void caml_spacetime_c_to_ocaml(void* ocaml_entry_point,
...
-extern void caml_array_bound_error(void);  /* fail.c */
```
Comment (dra27) [inline runtime/spacetime_nat.c:1000]: This is defined in `fail.h` (likewise already `#include`'d)

**Review by dra27 — COMMENTED** [review]
Notes on the tweaks

File: runtime/spacetime_nat.c (hunk 2 above)
Comment (xavierleroy) [reply runtime/spacetime_nat.c:132]: Giving a name to the enum would be written
```
enum features { ... };
```
Here we have an anonymous enum and a global variable `features` of type this enum.  Certainly not what was intended.

**Review by xavierleroy — APPROVED** [review]
Looks good to me.

**Pull request merged**